base/common/src/main/java/com/netscape/certsrv/profile/ProfileInput.java

```diff
@@ -206,18 +217,244 @@ public boolean equals(Object obj) {
+        for (int i = 0; i < configAttributeList.getLength(); i++) {
+            Element configAttributeElement = (Element) configAttributeList.item(i);
+            ProfileAttribute profileAttribute = new ProfileAttribute();
+            String configAttributeId = configAttributeElement.getAttribute("name");
+            profileAttribute.setName(configAttributeId);
+
+            NodeList configAttributeNameList = configAttributeElement.getElementsByTagName("Name");
+            if (configAttributeNameList.getLength() > 0) {
+                String name = configAttributeNameList.item(0).getTextContent();
+                profileAttribute.setName(name);
+            }
+
+            NodeList valueList = configAttributeElement.getElementsByTagName("Value");
+            if (valueList.getLength() > 0) {
+                String value = valueList.item(0).getTextContent();
+                profileAttribute.setValue(value);
+            }
+
+            NodeList descriptorList = configAttributeElement.getElementsByTagName("Descriptor");
+            if (descriptorList.getLength() > 0) {
+                String syntax = null;
+                String constraint = null;
+                String description = null;
+                String def = null;
+                NodeList syntaxList = configAttributeElement.getElementsByTagName("mSyntax");
+                NodeList mConstraintList = configAttributeElement.getElementsByTagName("mConstraint");
+                NodeList mDescriptionList = configAttributeElement.getElementsByTagName("mDescription");
+                NodeList defList = configAttributeElement.getElementsByTagName("mDef");
+                if (syntaxList.getLength() > 0) {
+                    syntax = syntaxList.item(0).getTextContent();
+                }
+                if (mConstraintList.getLength() > 0) {
+                    constraint = mConstraintList.item(0).getTextContent();
+                }
+                if (mDescriptionList.getLength() > 0) {
+                    description = mDescriptionList.item(0).getTextContent();
+                }
+                if (defList.getLength() > 0) {
+                    def = defList.item(0).getTextContent();
+                }
+                Descriptor descriptor = new Descriptor(syntax, constraint, def, description);
+                profileAttribute.setDescriptor(descriptor);
+            }
+            profileInput.addConfigAttribute(profileAttribute);
+         }
+        return profileInput;
+    }
+
     public String toXML() throws Exception {
-        Marshaller marshaller = JAXBContext.newInstance(ProfileInput.class).createMarshaller();
-        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.newDocument();
+
+        Element accountElement = toDOM(document);
```

Comment:
Maybe we should just call it `element`.

Reply:
Yeah I had a few instances where I simply forgot to rename it after I copied the `Account` change too.